### DISCUSSION TITLE
What Webhosting Services Would You Reccommend?

### DISCUSSION THREAD

**Discussion by arn5891** [Rapptz/discord.py#9493] [Q&A]
What sites do you use for bots? I originally planned to use webhosting services like Render and Back4App, but they timeout after a couple of hours. Is there any way to continuously host a bot without paid plans or providing credit card info?

**Comment by LostLuma** [comment]
There is generally no reputable hosting provider that you will find that has both a free tier and does not require a credit card. The only case in which both of these are true is if you are looking to make a slash-command-only bot, then you can use the services you mentioned or something like Cloudflare Workers, as Discord provides [outgoing HTTP webhooks for interactions](https://discord.com/developers/docs/interactions/receiving-and-responding#receiving-an-interaction) (note that you can not use discord.py for this).

A decent option without continuous payment is to either use an old computer or a device similar to a raspberry pi to host your bot at home, you can set it up without a screen to save power and have it work more like a real server.

Another would be to find a friend willing to host it, do note however that since they control the server they will (most likely) be able to access your data. Also remember to frequently take backups, since you do not know if this sort of situation is stable!

Note that there is also an [official Discord server](https://discord.gg/r3sSKJJ) for discord.py in which you can ask questions.

**Comment by No767** [comment]
I have always said this to others having the same exact question as you: **There are no such things as a "free" host**.

My recommendation would be either to pay up, or if you are an eligible student, then the deals on the Github Student Education Pack may help. Other than that, you are out of luck. As mentioned, there are two ways you can tackle this: self hosting or vps. Self hosting really means that you are running the server yourself. You set up a small homelab server or a raspberry pi and run your bot there. Using an VPS is the more preferred method since all you need to do is rent out a server from an VPS provider (eg Akamai/Linode, etc) and go from there. My preferred VPS provider is Hetzner and that VPS provider is a good place to start

**Comment by SeoulSKY** [comment]
I use Oracle Cloud for running my bot. It gives you **always free** 2 instances of a virtual machine (VM.Standard.E2.1.Micro) with 1 OCPU and 1GB of RAM and 1 virtual machine (VM.Standard.E3.Flex) with up to 4 OCPUs and 24GB of RAM if there is an empty one (this one is popular so most likely already used by others). It requires you to register a credit card (or debit card), but it will never charge you unless you manually upgrade your account to paid account.

If you don't prefer using Oracle Cloud, consider using Google Cloud. It gives you **always free** 1 instance of a virtual machine (e2-micro if I remember correctly) with 1 vCPU and 1GB of RAM

**Comment by arn5891** [comment]
Hi all, thanks for replying, I'm using Daki webhosting right now. it's free, not too well known, and has decent performance.
